tuya.cpp

```diff
@@ -228,31 +227,81 @@ void DeRestPluginPrivate::handleTuyaClusterIndication(const deCONZ::ApsDataIndic
+            quint16 heatSetpoint16;
 
-            quint8 part = 1;
+            quint8 part = 0;
             QList<int> listday;
-
-            if (dp == 0x0070) //work days (6)
-            {
-                listday << 124;
-            }
-            else if (dp == 0x00071) // holiday = Not working day (6)
-            {
-                listday << 003;
-            }
-            else if (dp == 0x0065) // Moe thermostat W124 (4) + W002 (4) + W001 (4)
+            
+            switch (dp)
             {
-                part = length / 3;
-                listday << 124 << 2 << 1;
-            }
+                case 0x0070: //work days (6)
+                {
+                    part = 1;
+                    listday << 124;
+                    length = length / 3;
+                }
+                break;
+                case 0x0071: // holiday = Not working day (6)
+                {
+                    part = 1;
+                    listday << 003;
+                    length = length / 3;
+                }
+                break;
+                case 0x0065: // Moe thermostat W124 (4) + W002 (4) + W001 (4)
+                {
+                    part = length / 3;
+                    listday << 124 << 2 << 1;
+                    length = length / 3;
+                }
+                break;
+                // Daily schedule (mode 8)(minut 16)(temperature 16)(minut 16)(temperature 16)(minut 16)(temperature 16)(minut 16)(temperature 16)
+                case 0x007B: // Sunday
+                case 0x007C: // Monday
+                case 0x007D: // Thuesday
+                case 0x007E: // Wednesday
+                case 0x007F: // Thursday
+                case 0x0080: // Friday
+                case 0x0081: // Saturday
+                {
+                    int t[8] = {1,64,32,46,8,4,2};
+                    part = 1;
+                    listday << t[dp-123];
+                    
+                    length = (length - 1) / 2;
+                    
+                    quint8 mode;
+                    stream >> mode; // First octet is the mode
+                    break;
 
+                }
+                default:
+                {
+                    DBG_Printf(DBG_INFO, "Tuya : Unknow Schedule mode\n");
+                }
+                break;
+            }
+            
             for (; part > 0; part--)
             {
                 for (; length > 0; length--)
                 {
-                    stream >> hour;
-                    stream >> minut;
-                    stream >> heatSetpoint;
+                    if (dp >= 0x007B && dp <= 0x0081)
+                    {
+                        stream >> minut16;
+                        stream >> heatSetpoint16;
+                        hour = (quint8) ((minut16 / 60) & 0xff);
+                        minut = (quint8) ((minut16 - 60 * hour) & 0xff);
+                        heatSetpoint = (quint8) ((heatSetpoint16 / 10) & 0xff);
```

Comment:
Always use C++11 casts instead old style C casts as they are nasty in terms of type safety and easily overlooked.

```cpp
hour = static_cast<quint8>((minut16 / 60) & 0xff);
minut = static_cast<quint8>((minut16 - 60 * hour) & 0xff);
heatSetpoint = static_cast<quint8>((heatSetpoint16 / 10) & 0xff);

```